src/reducers/index.js

```diff
@@ -0,0 +1,253 @@
+                store: action.transfer.from,
+                model: action.transfer.product,
+                inventory: -1 * action.transfer.quantity,
+                isNewInventory: false
+            },
+            {
+                store: action.transfer.to,
+                model: action.transfer.product,
+                inventory: action.transfer.quantity,
+                isNewInventory: false
+            }]);
+        default:
+            return state;
+    }
+};
+
+const transfer = (state = {
+    storeFrom: "",
+    storeTo: "",
+    shoeModel: "",
+    shoeQuantity: 0,
+    quantityFromOk: false,
+    quantityToOk: false
+}, action) => {
+    switch (action.type) {
+        case types.SET_STORE_FROM_TO:
+            if (action.store.type === "from") {
+                return {
+                    ...state,
+                    ...{ storeFrom: action.store.name }
+                }
+            } else {
+                return {
+                    ...state,
+                    ...{ storeTo: action.store.name }
+                }
+            }
+        case types.SET_SHOE_FROM_TO:
+            return {
+                ...state,
+                ...{
+                    shoeModel: action.shoeModel
+                }
+            }
+        case types.SET_QUANTITY:
+            return {
+                ...state,
+                ...{
+                    shoeQuantity: action.quantity
+                }
+            }
+        case types.SET_TRANSFER_VALIDITY:
+             if (action.transfer.type === "from") {
+                return {
+                    ...state,
+                    ...{ quantityFromOk: action.transfer.isValid }
+                }
+            } else {
+                return {
+                    ...state,
+                    ...{ quantityToOk: action.transfer.isValid }
+                }
+            }
+        case types.TRANSFER_RESET:
+            return {
+                ...state,
+                ...{
+                    storeFrom: "",
+                    storeTo: "",
+                    shoeModel: "",
+                    shoeQuantity: 0,
+                    quantityFromOk: false,
+                    quantityToOk: false
+                }
+            }
+        default:
+            return state;
+    }
+};
+
+export default combineReducers({
+    inventory,
+    visibilityFilter,
+    transfer
+});
+
+/**
+ * 
+ * @param {Object} state The inventory state
+ * @param {Array} entries array of inventory to edit.
+ *                      {
+ *                        store,
+ *                        model,
+ *                        inventory,
+ *                        isNewInventory
+ *                      }
+ */
+const alternInventory = (state, entries) => {
+    let shopsStatsCopy = Object.assign({}, state.shopsStats);
+    let aldoShop = state.shops.map(shop => {
+        let inventoryUpdate = false;
+        let shopCopy = Object.assign({}, shop);
```

Comment:
Copying every shop data on each action is not really efficient, do you now why and could you think of a better solution ?